### PR TITLE
New tutorial: manim internals basic

### PR DESCRIPTION
**Pull request by leotrs**
This adds a section that I felt was missing in the Quickstart tutorial, and adds a new tutorial "Manim internals: Basic".

The quickstart tutorial is meant to lead the user by the hand from installation to their first scene. The "A deeper look" tutorial explains the output folders and some common CLI arguments. The new tutorial is mean to explain the nature of mobjects, animations, and scenes, **not** with every detail, but just the common operations and how they relate to each other.

This is a work in progress, but I would very much appreciate reviews before the final thing is finished. Tagging some people who have shown interest in documentation: @kolibril13 @naveen521kk @Aathish04 @behackl

One thing: I kinda hate the title "Manim internals: Basic". I chose this because at first I thought there could be a few follow up tutorials "Manim internals: Mobject", "Manim internals: Animation", and "Manim internals: Scene", but now I'm not so sure. Thoughts?

### CODE REVIEW

**Comment by behackl** [comment]
This is very cool, wow!

I don't have a very strong opinion on the title; if you don't like "Manim internals: Basic" it could also simply be "Manim internals" -- or maybe something like "Manim internals: three basic concepts" / "Manim's building blocks: three basic concepts"?

**Comment by leotrs** [comment]
Yeah I think we should keep it simple. I'm thinking "Manim's building blocks" for this one, and then just have something like "More about Mobjects", "More about Animations", "More about Scenes" down the road.

**Comment by behackl** [comment]
In order to handle examples in a simpler way, it would be great to have #402 (and then use the manim directive instead of statically adding files).

File: docs/source/tutorials/building_blocks.rst
Comment (huguesdevimeux) [inline docs/source/tutorials/building_blocks.rst:237]: **Suggestion (if minor)** : Maybe add the way to set z-index (`set_z_index`).

**Review by huguesdevimeux — APPROVED** [review]
SVGTM !

File: docs/source/tutorials/building_blocks.rst
Comment (leotrs) [reply docs/source/tutorials/building_blocks.rst:237]: I tried, but the current implementation of `z_index` does not play well with `add` and `remove`. So the tutorial I _wanted_ to write is actually not possible at this moment.

File: docs/source/tutorials/building_blocks.rst
Comment (nilaybhatia) [inline docs/source/tutorials/building_blocks.rst:38]: Maybe hyperlink to a Wikipedia article on Vector Graphics: https://en.wikipedia.org/wiki/Vector_graphics#:~:text=Vector%20graphics%20are%20computer%20graphics,form%20polygons%20and%20other%20shapes.

File: docs/source/tutorials/building_blocks.rst
Comment (nilaybhatia) [inline docs/source/tutorials/building_blocks.rst:148]: Should we somewhere mention the default value for `next_to` is `RIGHT`?

Comment (leotrs) [reply docs/source/tutorials/building_blocks.rst:148]: Good suggestion!

**Comment by leotrs** [comment]
I think this is a good start and can be reviewed/merged now.

When reviewing, please note that the purpose of this tutorial is not to show every single feature of manim, but give a quick introduction to each of the three main classes (mobject, animation, scene).

As mentioned before, I think there should be a separate tutorial for each of these main classes that goes into detail of advanced usage. For example:

+ Mobject tutorial: VGroup, VDict, ImageMobject, PointCloudMobject,
+ Animation tutorial: AnimationGroup, LaggedStart, rate functions,
+ Scene tutorial: setup and tear_down, graphScene

**Comment by behackl** [comment]
Please, let us use #402 for including rendered examples (in order to keep the git history file size small). I even volunteer to adapt this PR as soon as #402 is merged. :-)

**Comment by leotrs** [comment]
Yes I agree, but this can be reviewed now. We can merge this right after 402.

**Review by nilaybhatia — APPROVED** [review]

File: docs/source/tutorials/building_blocks.rst
Comment (behackl) [inline docs/source/tutorials/building_blocks.rst:337]: Superfluous spaces.

```suggestion
to be :meth:`~.Scene.add`-ed to a scene to be displayed, or
:meth:`~.Scene.remove`-d from it to cease being displayed.  Every animation
has to be :meth:`~.Scene.play`-ed by a scene, and every time interval where no
```

File: docs/source/tutorials/quickstart.rst
Comment (behackl) [inline docs/source/tutorials/quickstart.rst:97]: Just a suggestion: These could be links to the reference manual.

```suggestion
the :meth:`~.Scene.construct` method of a class that derives from :class:`.Scene`.  Inside this
```

File: docs/source/tutorials/quickstart.rst
Comment (behackl) [inline docs/source/tutorials/quickstart.rst:107]: Again, just a suggestion:

```suggestion
Finally, the last line uses the animation :class:`.ShowCreation` to display the
```

File: docs/source/tutorials/quickstart.rst
Comment (behackl) [inline docs/source/tutorials/quickstart.rst:115]: Again, just a suggestion:

```suggestion
.. tip:: Every animation must be contained within the :meth:`~.Scene.construct` method of a
         class that derives from :class:`.Scene`.  Other code, for example auxiliary
```

File: docs/source/tutorials/building_blocks.rst
Comment (behackl) [inline docs/source/tutorials/building_blocks.rst:298]: The indentation is off here (and I am not confident enough in my counting skills to make a suggestion, sorry 🙃)

File: docs/source/tutorials/building_blocks.rst
Comment (behackl) [inline docs/source/tutorials/building_blocks.rst:323]: `IndentationError`

File: docs/source/tutorials/building_blocks.rst
Comment (behackl) [inline docs/source/tutorials/building_blocks.rst:262]: `IndentationError`

File: docs/source/tutorials/building_blocks.rst
Comment (leotrs) [reply docs/source/tutorials/building_blocks.rst:27]: Yes, this is intentional. Once those modules are added to the reference, these links will work automatically. I think there is no harm in leaving them for now :)

Note: those modules are currently not in the reference because their docstrings are less than great right now ;)

File: docs/source/tutorials/building_blocks.rst
Comment (leotrs) [reply docs/source/tutorials/building_blocks.rst:337]: I tried this, but I don't think it renders correctly. I think rst requires a space after each backtick `. Any workarounds?

File: docs/source/tutorials/building_blocks.rst
Comment (leotrs) [reply docs/source/tutorials/building_blocks.rst:300]: Good eye! Just fixed that in the latest commit.

**Comment by leotrs** [comment]
Ok, this is ready for merge. @behackl do we merge this now or wait until 402? Is 402 ready to review/merge?

File: docs/source/tutorials/building_blocks.rst
Comment (behackl) [reply docs/source/tutorials/building_blocks.rst:339]: Are you sure? In renders just fine if I try it locally.

Otherwise, its always possible to write custom text for the links, for example:

```
:meth:`added <.Scene.add>`
```

Comment (leotrs) [reply docs/source/tutorials/building_blocks.rst]: D'oh!

**Comment by behackl** [comment]
> Ok, this is ready for merge. @behackl do we merge this now or wait until 402? Is 402 ready to review/merge?

Excellent, this is fine for me now. I'd rather use the `.. manim::` directive from #402 to include the examples (especially as `some_animations.gif`, `apply_method.gif`, `run_time.gif` are currently not included here).

#402 is certainly ready to be reviewed (and IMO also ready to merge 😄).

File: docs/source/tutorials/building_blocks.rst
Comment (behackl) [inline docs/source/tutorials/building_blocks.rst:339]: When using custom text for the links, `~` destroys the links (because `~` is only used to shorten the automatically generated text).

```suggestion
<.Scene.remove>` from it to cease being displayed.  Every animation has to be
:meth:`played <.Scene.play>` by a scene, and every time interval where no
```

**Review by behackl — CHANGES_REQUESTED** [review]

File: docs/source/tutorials/building_blocks.rst
Comment (leotrs) [reply docs/source/tutorials/building_blocks.rst:339]: Good call!

File: docs/source/tutorials/building_blocks.rst
Comment (behackl) [reply docs/source/tutorials/building_blocks.rst:27]: Of course, linking them nonetheless is the correct choice.

I would argue, however, that no matter how non-existant the docstrings for these modules are, they should still be included so that the classes can be found in the documentation. The reference manual should be, IMO, as complete as possible.

**Pull request merged**
